pkg/messageformat/lex.go

```diff
@@ -370,3 +371,18 @@ func isNameChar(v rune) bool {
 func isSpace(r rune) bool {
 	return r == ' ' || r == '\t' || r == '\r' || r == '\n'
 }
+
+func processFunction(l *lexer, v rune) stateFn {
+	if l.exprDepth > 1 {
+		switch v {
+		case '+':
+			return lexOpeningFunction(l)
+		case '-':
+			return lexClosingFunction(l)
+		}
+	}
+
+	l.backup()
+
+	return lexText(l)
+}
```

Comment:
... or pass the lex function

```suggestion
func processFunction(l *lexer, f ...) stateFn {
	if l.exprDepth > 1 {
		f(l)
	}

	l.backup()

	return lexText(l)
}
```